### PR TITLE
Improvements for proposal token voting

### PR DESCRIPTION
**Pull request by Devorein**
### WHAT

Please read commit messages!

### CODE REVIEW

File: lib/proposals/createProposal.ts [hunk 1]
```diff
@@ -240,10 +240,13 @@ export async function createProposal({
       ? await upsertProposalFormAnswers({ proposalId, answers: formAnswers })
       : null;
 
-  await createVoteIfNecessary({
-    createdBy: userId,
-    proposalId
-  });
+  // make sure proposal is published before we create a vote
+  if (!isDraft) {
```
Comment (Devorein) [inline lib/proposals/createProposal.ts:244]: This was a bug, if your first step was a vote and you first create a draft then published it, it threw an error

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Settings/components/EvaluationStepSettings.tsx [hunk 2]
```diff
@@ -54,6 +60,15 @@ export function EvaluationStepSettings({
     });
   }
 
+  useEffect(() => {
+    if (evaluation.type === 'vote' && evaluation.voteSettings?.strategy === 'token') {
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Settings/components/EvaluationStepSettings.tsx:64]: suggestion: just define this outside:
```
const isTokenVoting = evaluation.type === 'vote' && evaluation.voteSettings?.strategy === 'token'
```
and use isTokenVoting as a dependency. i don't think u need an extra state

File: lib/votes/getVotingPowerForVotes.ts [hunk 3]
```diff
@@ -20,14 +20,29 @@ export async function getVotingPowerForVotes({
     }
   });
 
+  const user = await prisma.user.findUniqueOrThrow({
```
Comment (mattcasey) [inline lib/votes/getVotingPowerForVotes.ts:23]: suggestion: request the user's wallets here too, instead of doing two queries to prisma

**Review by mattcasey — APPROVED** [review]

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Settings/components/EvaluationStepSettings.tsx (hunk 2 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalEvaluations/components/Settings/components/EvaluationStepSettings.tsx:64]: Updated.

File: lib/votes/getVotingPowerForVotes.ts (hunk 3 above)
Comment (Devorein) [reply lib/votes/getVotingPowerForVotes.ts:23]: Good one. Updated.

**Pull request merged**